### ISSUE TITLE
gg_season and gg_subseries generating vctrs warnings

### ISSUE THREAD

**Issue by robjhyndman** [tidyverts/feasts#123]
``` r
library(tsibble)
library(dplyr)
#> 
#> Attaching package: 'dplyr'
#> The following objects are masked from 'package:stats':
#> 
#>     filter, lag
#> The following objects are masked from 'package:base':
#> 
#>     intersect, setdiff, setequal, union
library(feasts)
#> Loading required package: fabletools
tsibbledata::aus_retail %>%
  filter(
    State == "Victoria",
    Industry == "Cafes, restaurants and catering services"
  ) %>%
  gg_season(Turnover)
#> Warning in NextMethod("["): Incompatible methods (">=.Date", ">=.vctrs_vctr")
#> for ">="
#> Warning in NextMethod("["): Incompatible methods ("<=.Date", "<=.vctrs_vctr")
#> for "<="
#> Warning in NextMethod("["): Incompatible methods (">=.Date", ">=.vctrs_vctr")
#> for ">="
#> Warning in NextMethod("["): Incompatible methods ("<=.Date", "<=.vctrs_vctr")
#> for "<="
```

![](https://i.imgur.com/k5GrpYp.png)

<sup>Created on 2020-12-23 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>
<details>
<summary>
Session info
</summary>

``` r
sessionInfo()
#> R version 4.0.3 (2020-10-10)
#> Platform: x86_64-pc-linux-gnu (64-bit)
#> Running under: Ubuntu 20.10
#> 
#> Matrix products: default
#> BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0
#> LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.9.0
#> 
#> locale:
#>  [1] LC_CTYPE=en_AU.UTF-8       LC_NUMERIC=C              
#>  [3] LC_TIME=en_AU.UTF-8        LC_COLLATE=en_AU.UTF-8    
#>  [5] LC_MONETARY=en_AU.UTF-8    LC_MESSAGES=en_AU.UTF-8   
#>  [7] LC_PAPER=en_AU.UTF-8       LC_NAME=C                 
#>  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
#> [11] LC_MEASUREMENT=en_AU.UTF-8 LC_IDENTIFICATION=C       
#> 
#> attached base packages:
#> [1] stats     graphics  grDevices datasets  utils     methods   base     
#> 
#> other attached packages:
#> [1] feasts_0.1.6.9000     fabletools_0.2.1.9000 dplyr_1.0.2          
#> [4] tsibble_0.9.3        
#> 
#> loaded via a namespace (and not attached):
#>  [1] Rcpp_1.0.5                pillar_1.4.7             
#>  [3] compiler_4.0.3            highr_0.8                
#>  [5] tools_4.0.3               digest_0.6.27            
#>  [7] lubridate_1.7.9.2         evaluate_0.14            
#>  [9] lifecycle_0.2.0           tibble_3.0.4             
#> [11] gtable_0.3.0              anytime_0.3.9            
#> [13] pkgconfig_2.0.3           rlang_0.4.9              
#> [15] curl_4.3                  yaml_2.2.1               
#> [17] xfun_0.19                 xml2_1.3.2               
#> [19] httr_1.4.2                stringr_1.4.0            
#> [21] knitr_1.30                generics_0.1.0           
#> [23] vctrs_0.3.6.9000          grid_4.0.3               
#> [25] tidyselect_1.1.0          glue_1.4.2               
#> [27] R6_2.5.0                  distributional_0.2.1.9000
#> [29] rmarkdown_2.6             tsibbledata_0.2.0        
#> [31] purrr_0.3.4               farver_2.0.3             
#> [33] ggplot2_3.3.2             tidyr_1.1.2              
#> [35] magrittr_2.0.1            scales_1.1.1             
#> [37] ellipsis_0.3.1            htmltools_0.5.0          
#> [39] mime_0.9                  colorspace_2.0-0         
#> [41] renv_0.12.3               labeling_0.4.2           
#> [43] stringi_1.5.3             munsell_0.5.0            
#> [45] crayon_1.3.4
```

</details>

**Comment by robjhyndman** [comment]
I think this fix has led to other issues:

``` r
library(fpp3)
#> ── Attaching packages ─────────────────────────────────────── fpp3 0.3.0.9000 ──
#> ✓ tibble      3.0.4          ✓ tsibble     0.9.3     
#> ✓ dplyr       1.0.2          ✓ tsibbledata 0.2.0     
#> ✓ tidyr       1.1.2          ✓ feasts      0.1.6.9000
#> ✓ lubridate   1.7.9.2        ✓ fable       0.2.1.9000
#> ✓ ggplot2     3.3.2
#> ── Conflicts ───────────────────────────────────────────────── fpp3_conflicts ──
#> x lubridate::date()   masks base::date()
#> x dplyr::filter()     masks stats::filter()
#> x tsibble::interval() masks lubridate::interval()
#> x dplyr::lag()        masks stats::lag()
vic_elec %>% gg_season(Demand, period = "day") 
#> Error: Can't convert <datetime<UTC>> to <datetime<UTC>>.
```

![](https://i.imgur.com/APV9P0u.png)

<sup>Created on 2020-12-25 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by mitchelloharawild** [comment]
Fixed in 68a3602232ce72b5111ccc7bf5af5ee3e6ae439a

``` r
library(fpp3)
#> ── Attaching packages ────────────────────────────────────────────── fpp3 0.3 ──
#> ✓ tibble      3.0.4          ✓ tsibble     0.9.4     
#> ✓ dplyr       1.0.2          ✓ tsibbledata 0.2.0     
#> ✓ tidyr       1.1.2          ✓ feasts      0.1.6.9000
#> ✓ lubridate   1.7.9.2        ✓ fable       0.2.1.9000
#> ✓ ggplot2     3.3.3
#> ── Conflicts ───────────────────────────────────────────────── fpp3_conflicts ──
#> x lubridate::date()    masks base::date()
#> x dplyr::filter()      masks stats::filter()
#> x tsibble::intersect() masks base::intersect()
#> x tsibble::interval()  masks lubridate::interval()
#> x dplyr::lag()         masks stats::lag()
#> x tsibble::setdiff()   masks base::setdiff()
#> x tsibble::union()     masks base::union()
vic_elec %>% gg_season(Demand, period = "day") 
```

![](https://i.imgur.com/MbXlPCa.png)

<sup>Created on 2021-01-11 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>
